wallet/src/de/schildbach/wallet/service/platform/work/RestoreIdentityWorker.kt

```diff
@@ -293,18 +293,31 @@ class RestoreIdentityWorker @AssistedInject constructor(
                     }
                 }
 
-                platformRepo.updateIdentityCreationState(blockchainIdentityData, BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKED)
-                platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
-
-                platformRepo.updateIdentityCreationState(blockchainIdentityData, BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKING)
-
-                // recover the verification link
-
-                platformRepo.updateIdentityCreationState(blockchainIdentityData, BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKED)
-                platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
-
-                platformRepo.updateIdentityCreationState(blockchainIdentityData, BlockchainIdentityData.CreationState.VOTING)
-                platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
+                if (blockchainIdentity.currentUsername != null) {
+
+                    platformRepo.updateIdentityCreationState(
+                        blockchainIdentityData,
+                        BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKED
+                    )
+                    platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
+                    platformRepo.updateIdentityCreationState(
+                        blockchainIdentityData,
+                        BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKING
+                    )
+
+                    // recover the verification link
+                    platformRepo.updateIdentityCreationState(
+                        blockchainIdentityData,
+                        BlockchainIdentityData.CreationState.REQUESTED_NAME_CHECKED
+                    )
+                    platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
+                    // set voting state
+                    platformRepo.updateIdentityCreationState(
+                        blockchainIdentityData,
+                        BlockchainIdentityData.CreationState.VOTING
+                    )
+                    platformRepo.updateBlockchainIdentityData(blockchainIdentityData, blockchainIdentity)
+                }
```

Comment:
This will prevent a crash or other errors when restoring an identity that doesn't have a username.